File: AD-HOC/PYTHON/1026_1147/1032.py
def josephus(n):
    retorno = 0
    for i in range(1, n + 1):
        retorno = (retorno + primos[n - i]) % i
    return retorno + 1

def is_prime(x):
    if x == 2:
        return True
    if x % 2 == 0 or x < 2:
        return False
    for i in range(3, int(x**0.5) + 1, 2):
        if x % i == 0:
            return False
    return True

def preenche_primo():
    primos = []
    i = 2
    while len(primos) < MAXSIZE:
        if is_prime(i):
            primos.append(i)
        i += 1
    return primos

MAXSIZE = 3502
MAXPRIME = 32650
primos = preenche_primo()

while True:
    n = int(input())
    if n == 0:
        break
    print(josephus(n))

File: AD-HOC/C#/TELA_1/1032.cs
using System;

class Program
{
    const int MAXSIZE = 3502;
    const int MAXPRIME = 32650;
    static int[] primos = new int[MAXSIZE];

    static void Main()
    {
        PreenchePrimo();
        while (true)
        {
            int n = int.Parse(Console.ReadLine());
            if (n == 0)
                break;
            Console.WriteLine(Josephus(n));
        }
    }

    static int Josephus(int n)
    {
        int retorno = 0;
        for (int i = 1; i <= n; ++i)
            retorno = (retorno + primos[n - i]) % i;
        return retorno + 1;
    }

    static bool IsPrime(int x)
    {
        if (x == 2)
            return true;
        if (x % 2 == 0 || x < 2)
            return false;
        for (int i = 3; i * i <= x; i += 2)
            if (x % i == 0)
                return false;
        return true;
    }

    static void PreenchePrimo()
    {
        for (int i = 2, j = 0; j < MAXSIZE; ++i)
        {
            if (IsPrime(i))
                primos[j++] = i;
        }
    }
}
